amr-wind/wind_energy/ABLBoundaryPlane.cpp

```diff
@@ -0,0 +1,624 @@
+#include "amr-wind/CFDSim.H"
+#include "amr-wind/wind_energy/ABLBoundaryPlane.H"
+#include "amr-wind/wind_energy/ABLFillInflow.H"
+#include "AMReX_Gpu.H"
+#include "AMReX_ParmParse.H"
+#include "amr-wind/utilities/ncutils/nc_interface.H"
+
+namespace amr_wind {
+
+namespace {
+
+//! Return closest index (from lower) of value in vector
+AMREX_FORCE_INLINE int
+closest_index(const amrex::Vector<amrex::Real>& vec, const amrex::Real value)
+{
+    auto const it = std::upper_bound(vec.begin(), vec.end(), value);
+    AMREX_ALWAYS_ASSERT(it != vec.end());
+
+    const int idx = std::distance(vec.begin(), it);
+    return std::max(idx - 1, 0);
+}
+
+//! Return indices perpendicular to normal
+AMREX_FORCE_INLINE amrex::GpuArray<int, 2>
+get_perpendicular_idx(const int normal)
```

Comment:
Given what is happening in line 247 You might want this as 

```C++
template <typename T = amrex::GpuArray<int, 2>>
T get_perpendicular_idx(const int normal);
```

Reply:
fixed